compiler/rustc_const_eval/src/const_eval/mod.rs

```diff
@@ -39,23 +44,15 @@ pub(crate) fn const_caller_location(
     ConstValue::Scalar(Scalar::from_maybe_pointer(loc_place.ptr, &tcx))
 }
 
-// We forbid type-level constants that contain more than `VALTREE_MAX_NODES` nodes.
-const VALTREE_MAX_NODES: usize = 1000;
```

Comment:
> Using some arbitrary threshold might also break crates.

not on stable, on stable all valtrees have a size of 1 because only scalars are allowed. Getting a weird error is preferable to getting an ICE in these cases